backend/data.go

```diff
@@ -134,3 +158,126 @@ type TimeRange struct {
 func (tr TimeRange) Duration() time.Duration {
 	return tr.To.Sub(tr.From)
 }
+
+type DataResponseType string
+
+const (
+	DataResponseTypeArrow DataResponseType = "ARROW"
```

Comment:
This seems good for clarity, but I *think* we could get the same thing with testing first byte for `{`

Reply:
Agree, but we'd rather be explicit and structured around this. We don't know what happens in the future and whether we need to introduce changes to the JSON structure, introduce a new data format etc why keeping this makes more sense. 

The key is capturing the user/plugin developer intent so that we for sure knows which format has been returned without "guessing".